### ISSUE TITLE
Detect when ArgoCD is misbehaving and surface that to the web interface

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#14145]
# Summary

I have some ArgoCD instances that are apparently trying to generate manifests near constantly, and apparently ArgoCD isn't supposed to do that.

ArgoCD should have certain conditions that it recognizes as "bad" and reflect that in the web interface.

The simplest one is "refreshing an application more than once every X".

# Motivation

This reduces costs for everyone (cpu, logging, network)

# Proposal

Some metrics and thresholds and triggers in the reposerver/controller plus whatever bits are necessary to enable the web interface to show that things have reached this bad state.

**Comment by andrii-korotkov-verkada** [comment]
I like the idea, yet have a different opinion on the conditions to use.

Refreshing frequently might be okay for large applications with many pods, since there can be a lot of events generated by Kubernetes and we want to refresh, in particular to make sure syncs are progressing fast.

I think of specifying when cpu and memory usage is above requests, though maybe external alerting is supposed to solve that. Another one could be if there are many applications in the Unknown state.

**Comment by agaudreault** [comment]
I think you can use the current sync metrics to detect the frequency of sync. This is a known problem when auto-sync is on, and apps are constantly having diff due to mutation webhooks or other controllers making changes. 

I think this is a problem that should be fixed with Metrics, and argo already has feature that allows you to send metrics-based alerts to the application owners (you can define the owner as a label on the apps that will end up in the metric system).

You can also [kube-state-metrics for the ArgoCD custom resources](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/extend/customresourcestate-metrics.md) and have a metric based on the argo`.status.history.deployedAt` field. 

I will close this issue since I think an alert will be more valuable when an application is in that state than a visual indicator warning on the app web interface. Let me know if you disagree :) 

If you or anyone else in the community have an alert on "refreshing an application more than once every X", feel free to share

**Comment by alexymantha** [comment]
I agree that this makes more sense to monitor through metrics. It is also arguably already shown in the UI by the fact the the app is constantly syncing, though not directly/explicitly.

For future viewers of this issue: here is a pretty basic PromQL query to detect apps that sync often: ```
sum(round(increase(argocd_app_sync_total{phase="Succeeded"}[10m]))) by (name) * on(name) group_right() argocd_app_labels > 20
```
`20` being an arbitrary number but it has worked well so far.
`group_right()` with `argocd_app_labels` makes all labels on the app available in the result. Assuming you have information about the owner of the application in the labels, the info can then be used to route the alert accordingly.
